src/bundles/programmable_repl/functions.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * Functions for Programmable REPL
+ * @module programmable_repl
+ * @author Wang Zihan
+ */
+
+
+import { context } from 'js-slang/moduleHelpers';
+import { ProgrammableRepl } from './programmable_repl';
+
+const INSTANCE = new ProgrammableRepl();
+context.moduleContexts.programmable_repl.state = INSTANCE;
+/**
+ * Setup the programmable REPL with given metacircular evaulator entrance function
+ * @param {evalFunc} evalFunc - metacircular evaulator entrance function
+ *
+ * @category Main
+ */
+export function invoke_repl(evalFunc: Function) {
+  if (!(evalFunc instanceof Function)) {
+    const typeName = typeof (evalFunc);
+    throw new Error(`Wrong parameter type "${typeName}' in function "invoke_repl". It supposed to be a function and it's the entrance function of your metacircular evaulator.`);
+  }
+  INSTANCE.evalFunction = evalFunc;
+  return {
+    toReplString: () => '<Programmable REPL Initialized>',
+  };
+}
+
+
+/**
+ * Redirects the display message into Programmable REPL Tab and returns the string parameter
+ * @param {content} the content you want to display
+ *
+ * @category Main
+ */
+export function module_display(content : any) : any {

```

Comment:
This may be better named as `display_in_repl` or something to that effect. Our convention is to name methods with verbs, and `module_display` doesn't quite convey what this method is used for